unitycatalog-ai/integrations/autogen/README.md

```diff
@@ -0,0 +1,197 @@
+# Using Unity Catalog AI with the Autogen SDK
+
+You can use the Unity Catalog AI package with the autogen SDK to utilize functions that are defined in Unity Catalog to be used as tools within autogen LLM calls.
+
+> [!NOTE]
+> Ensure that the base Autogen package is installed with version `autogen-agentchat~=0.2` or earlier, as there are significant changes in the API after this release..
+
+
+## Installation
+
+```sh
+# install from the source
+pip install git+https://github.com/puneet-jain159/unitycatalog.git@autogen_ucai#subdirectory=unitycatalog-ai/integrations/autogen
+```
+
+> [!NOTE]
+> Once this package is published to PyPI, users can install via `pip install unitycatalog-autogen`
+
+## Get started
+
+### Databricks-managed UC
+
+To use Databricks-managed Unity Catalog with this package, follow the [instructions](https://docs.databricks.com/en/dev-tools/cli/authentication.html#authentication-for-the-databricks-cli) to authenticate to your workspace and ensure that your access token has workspace-level privilege for managing UC functions.
+
+#### Client setup
+
+Initialize a client for managing UC functions in a Databricks workspace, and set it as the global client.
+
+```python
+from ucai.core.client import set_uc_function_client
+from ucai.core.databricks import DatabricksFunctionClient
+
+client = DatabricksFunctionClient(
+    warehouse_id="..." # replace with the warehouse_id
+    cluster_id="..." # optional, only pass when you want to use cluster for function creation
+)
+
+# sets the default uc function client
+set_uc_function_client(client)
+```
+
+#### Create a function in UC
+
+Create a python UDF in Unity Catalog with the client
```

Comment:
let's show the higher level API `create_python_function` to simplify this example, since we're using a simply python function here